Packages/ClientRuntime/Sources/Networking/Http/CRT/CRTClientEngine.swift

```diff
@@ -37,7 +37,10 @@ public class CRTClientEngine: HttpClientEngine {
             logger.error("Server name was not able to be set in TLS Connection Options. TLS Negotiation will fail.")
             logger.error("Error: \(err.localizedDescription)")
         }
-        let socketOptions = SocketOptions(socketType: .stream)
+        var socketOptions = SocketOptions(socketType: .stream)
+#if os(iOS) || os(watchOS)
+        socketOptions.connectTimeoutMs = 30_000
+#endif
```

Comment:
I'm not quite sure if i understand the need to do this if we are doing it in the crt:
https://github.com/awslabs/aws-crt-swift/pull/56/files

Reply:
chatted offline-- other than changing the commit title to something like "tcp connection timeout", lgtm!

Reply:
answered offline but basically this is a different timeout. this timeout is tcp and the other is tls negotiation timeout.